pkg/component/controller/etcd.go

```diff
@@ -195,6 +195,14 @@ func (e *Etcd) Start(ctx context.Context) error {
 		args["--auth-token"] = auth
 	}
 
+	for name, value := range e.Config.ExtraArgs {
+		argName := fmt.Sprintf("--%s", name)
+		if args[argName] != "" {
```

Comment:
Better use ok in case we have valueless flags?

```suggestion
		if _, ok := args[argName]; ok {
```

Reply:
That's a very valid point, since you noticed this I'll also fix the other three instances where this happens.